std/parallelism.d

```diff
@@ -945,11 +915,74 @@ if (is(typeof(fun(args))) && isSafeTask!F)
     return ret;
 }
 
+version(useSysctlbyname)
+    private extern(C) int sysctlbyname(
+        const char *, void *, size_t *, void *, size_t
+    ) @nogc nothrow;
+
 /**
 The total number of CPU cores available on the current machine, as reported by
 the operating system.
 */
-immutable uint totalCPUs;
+@property @nogc nothrow pure @trusted uint totalCPUs()
+{
+    return (cast(uint function() @nogc nothrow pure)
+        &totalCPUsImpl)();
+}
+
+uint totalCPUsImpl() @nogc nothrow
+{
+    static shared uint result;
+    auto localResult = atomicLoad(result);
+    if (localResult > 0) return localResult;
+
+    // There might be harmless multiple initialization here
+    version(Windows)
+    {
+        // BUGS:  Only works on Windows 2000 and above.
+        import core.sys.windows.windows : SYSTEM_INFO, GetSystemInfo;
+        import std.algorithm.comparison : max;
+
+        SYSTEM_INFO si;
+        GetSystemInfo(&si);
+        atomicStore(result, max(1, cast(uint) si.dwNumberOfProcessors));
+    }
+    else version(linux)
+    {
+        import core.sys.posix.unistd : _SC_NPROCESSORS_ONLN, sysconf;
+        atomicStore(result, cast(uint) sysconf(_SC_NPROCESSORS_ONLN));
+    }
+    else version(Solaris)
+    {
+        import core.sys.posix.unistd : _SC_NPROCESSORS_ONLN, sysconf;
+        atomicStore(result, cast(uint) sysconf(_SC_NPROCESSORS_ONLN));
+    }
+    else version(useSysctlbyname)
+    {
+        version(OSX)
+        {
+            auto nameStr = "machdep.cpu.core_count\0".ptr;
+        }
+        else version(FreeBSD)
+        {
+            auto nameStr = "hw.ncpu\0".ptr;
+        }
+        else version(NetBSD)
+        {
+            auto nameStr = "hw.ncpu\0".ptr;
+        }
+
+        localResult = 0;
+        size_t len = uint.sizeof;
```

Comment:
Use `const len` instead of `size_t len`

Reply:
doesn't work, see signature: https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man3/sysctlbyname.3.html